catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/adapter/BrickAdapter.kt

```diff
@@ -0,0 +1,441 @@
+    private var checkBoxMode = NONE
+
+    private var scripts: MutableList<Script> = ArrayList()
+    private var firstConnectedItem = -1
+    private var lastConnectedItem = -1
+
+    private val selectionManager = MultiSelectionManager()
+    private val viewStateManager = ViewStateManager()
+
+    private var onItemClickListener: OnItemClickListener? = null
+    private var selectionListener: SelectionListener? = null
+
+    val items: MutableList<Brick> = ArrayList()
+
+    init {
+        updateItems(sprite)
+    }
+
+    companion object {
+        const val DISABLED_BRICK_ALPHA = .8f
+        const val NONE = 0
+        const val ALL = 1
+        const val SCRIPTS_ONLY = 2
+        const val CONNECTED_ONLY = 3
+
+        @JvmStatic
+        fun colorAsCommentedOut(background: Drawable) {
+            val matrix = ColorMatrix()
+            matrix.setSaturation(0f)
+            val filter = ColorMatrixColorFilter(matrix)
+            background.mutate()
+            background.colorFilter = filter
+        }
+    }
+
+    fun setOnItemClickListener(onItemClickListener: OnItemClickListener?) {
+        this.onItemClickListener = onItemClickListener
+    }
+
+    fun setSelectionListener(selectionListener: SelectionListener?) {
+        this.selectionListener = selectionListener
+    }
+
+    fun setCheckBoxMode(checkBoxMode: Int) {
+        this.checkBoxMode = checkBoxMode
+        notifyDataSetChanged()
+    }
+
+    fun updateItems(sprite: Sprite?) {
+        sprite?.scriptList?.let { scripts = it }
+        updateItemsFromCurrentScripts()
+    }
+
+    private fun updateItemsFromCurrentScripts() {
+        items.clear()
+        sprite.removeAllEmptyScriptBricks()
+        for (script in scripts) {
+            script.setParents()
+            script.addToFlatList(items)
+        }
+        notifyDataSetChanged()
+    }
+
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
+        val item = items[position]
+        val itemView = item.getView(parent.context)
+
+        itemView.visibility =
+            if (viewStateManager.isVisible(position)) View.VISIBLE else View.INVISIBLE
+        itemView.alpha = if (viewStateManager.isEnabled(position)) 1F else DISABLED_BRICK_ALPHA
+
+        var brickViewContainer = (itemView as ViewGroup).getChildAt(1)
+        if (item is UserDefinedReceiverBrick) {
+            brickViewContainer = (itemView.getChildAt(1) as ViewGroup).getChildAt(0)
+        }
+
+        val background = brickViewContainer.background
+        if (item.isCommentedOut || item is EmptyEventBrick) {
+            colorAsCommentedOut(background)
+        } else {
+            background.clearColorFilter()
+        }
+
+        checkBoxClickListener(item, itemView, position)
+        item.checkBox.isChecked = selectionManager.isPositionSelected(position)
+        item.checkBox.isEnabled = viewStateManager.isEnabled(position)
+        return itemView
+    }
+
+    @Suppress("ComplexMethod")
```

Comment:
maybe create a method to make it less complex

Reply:
@Walcho1125 The method has been split up, the suppress has been removed.